### ISSUE TITLE
Unable to use selectattr | difference to exclude a set of files from ordered list

### ISSUE THREAD

**Issue by cognitiaclaeves** [ansible/ansible#24010]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
selectattr | difference

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = /home/Q2DC/jnorment/gitrepo/ansible-playbooks/ansible-q2smart-ds-endpoints/ansible.cfg  configured module search path = ['library']
  python version = 3.5.1 (default, Oct 21 2016, 21:37:19) [GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
```

##### SUMMARY
Attempting to identify the latest X directories ( according to mtime ), while also skipping a set of known directories.

##### STEPS TO REPRODUCE

```yaml
- hosts: 127.0.0.1
  connection: local

  vars:
    - temp_path: '/tmp/temp_dir'
    - dirs_to_ignore:
       - ignore_this_one
       - ignore_this_one_too
    - dirs_to_make: [ one, two, three, four, five, six, seven ]
    - versions_to_keep: 4

  tasks:

    - name: Create directories
      file:
        path: "{{ temp_path }}/{{ item }}/"
        state: directory
      with_items: "{{ dirs_to_make }}"

    - name: Create directories to ignore
      file:
        path: "{{ temp_path }}/{{ item }}/"
        state: directory
      with_items: "{{ dirs_to_ignore }}"

    - name: Get potential versions
      find:
          paths: '{{ temp_path }}'
          file_type: directory
      register: potential_versions

    - name: Setfact for directories, no softlinks
      set_fact:
        # Put list of potential_versions into variable, sorted by reverse mtime, if not a soft-link
        potential_versions_no_links: "{{ potential_versions.files|selectattr('islnk', 'equalto', false)|sort(attribute='mtime', reverse=True)|map(attribute='path')|list }}"


    - name: Setfact for directories, no softlinks, sans directories to skip
      set_fact:
        # Put list of potential_versions into variable, sorted by reverse mtime, if not a soft-link
        potential_versions_no_skips: "{{ potential_versions.files|selectattr('islnk', 'equalto', false)|sort(attribute='mtime', reverse=True)|map(attribute='path')|difference(dirs_to_ignore)|list }}"

    - name: Show list
      debug:
        var: potential_versions_no_links

    # Expect to skip ignore_this_one_too, ignore_this_one, seven, & six
    # ( so five - one shoule be in list )
    - name: Show slice of bottom of list ( skipping {{ versions_to_keep }} elements )
      debug:
        var: potential_versions_no_links[versions_to_keep:]

    # Expect to skip seven, six, five, four, and not have ignore_this_one* in the list
    # ( so three, two, one should be in list )
    # *** This was not the case
    - name: Show same slice of bottom of list, sans the list in dirs_to_ignore
      debug:
        var: potential_versions_no_skips[versions_to_keep:]
```

##### EXPECTED RESULTS

```
TASK [Show same slice of bottom of list, sans the list in dirs_to_ignore] ****************************
ok: [127.0.0.1] => {
    "changed": false,
    "potential_versions_no_links[versions_to_keep:]": [
        "/tmp/temp_dir/three",
        "/tmp/temp_dir/two",
        "/tmp/temp_dir/one"
    ]
}
```

##### ACTUAL RESULTS

```
TASK [Show same slice of bottom of list, sans the list in dirs_to_ignore] ****************************
ok: [127.0.0.1] => {
    "changed": false,
    "potential_versions_no_skips[versions_to_keep:]": [
        "/tmp/temp_dir/three",
        "/tmp/temp_dir/four",
        "/tmp/temp_dir/ignore_this_one",
        "/tmp/temp_dir/two",
        "/tmp/temp_dir/one"
    ]
}
```

**Comment by sivel** [comment]
List Information
================

Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

This appears to be a user question, and we'd like to direct these kinds of things to either the mailing list or the IRC channel.

   * IRC: #ansible on irc.freenode.net   
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

If you can stop by there, we'd appreciate it.  This allows us to keep the issue tracker for bugs, pull requests, RFEs and the like.

Thank you once again and we look forward to seeing you on the list or IRC.  Thanks!

**Comment by cognitiaclaeves** [comment]
Why is this considered a "user question" ? I've found irc to be hit-or-miss, and it's not allowed at work. I didn't consider the mailing list b/c that just seemed to me to be antiquated, but I can see how that was an assumption on my part. This was the first time I've tried to 'give something back' to the Ansible community, and I'm a little upset that I put all of that work in and just had the issue closed. I'm partly wondering why I didn't see that my inquiry wouldn't be welcome here as a "user question" when I took the time to fill out the bug report.

Did I miss something in that process? Am I to take it, then, that in the future, I should ask my question in the mailing list, unless told explicitly to file a bug report? Or maybe Ansible's bar for participation is just too high for my disorganized lifestyle.
